### ISSUE TITLE
Do not highlight legacy APIs as deprecated, because they are not

### ISSUE THREAD

**Issue by mcollina** [neostandard/neostandard#183]
Currently neostandard is highlighting `process.nextTick()` has deprecated, while it's legacy and really not a problem to use it.

https://github.com/nodejs/undici/pull/3707/files#file-test-client-pipeline-js-L245

<img width="1167" alt="Screenshot 2024-10-09 at 12 48 24" src="https://github.com/user-attachments/assets/d8e8afe0-2313-4cf1-ba7e-6d5e3692e5a8">

**Comment by jerome-benoit** [comment]
It's not related to neostandard: the API definition is flagging it wrongly as 'deprecated'.

**Comment by mcollina** [comment]
Here is it: https://github.com/eslint-community/eslint-plugin-n/pull/350.

Do you know if there is a way to disable that specifically? This is appearing _everywhere_, and it's not possible to migrate off.

**Comment by jerome-benoit** [comment]
> Do you know if there is a way to disable that specifically? This is appearing _everywhere_, and it's not possible to migrate off.

Since the rule https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-deprecated-api.md does not seem to take any options permitting a fine-grained deprecation disablement, your best bet is to disable it entirely in the eslint configuration until a solution is found upstream.

**Comment by jerome-benoit** [comment]
Or maybe: 

```json
...
        "n/no-deprecated-api": ["error", {
            "version": "<22.7.0",
        }]
...
```

(untested)

**Comment by scagood** [comment]
This was a misunderstanding on my part, made a fix here:
https://github.com/eslint-community/eslint-plugin-n/pull/351

---

I will also make an issue for `fine-grained deprecation disablement` as this is a great idea!

**Comment by scagood** [comment]
Fix going out in [17.11.1](https://github.com/eslint-community/eslint-plugin-n/releases/tag/v17.11.1)
